cdap-security/src/main/java/io/cdap/cdap/security/auth/UserIdentityPair.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright © 2020-2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.security.auth;
+
+/**
+ * Contains a {@link UserIdentity} backed by a user credential in String form.
+ */
+public class UserIdentityPair {
+  private final String userCredential;
+  private final UserIdentity userIdentityObj;
+
+  public UserIdentityPair(String userCredential, UserIdentity userIdentityObj) {
+    this.userIdentityObj = userIdentityObj;
+    this.userCredential = userCredential;
+  }
+
+  public String getUserCredential() {
```

Comment:
Can you please add a javadoc on what this field contains? From the code it's something base64-encoded, but it's not clear for the user.

Reply:
Done. Just to clarify, this field can contain credentials of any form; it does not necessarily need to be of a particular format. For the AccessTokenIdentityExtractor, it so happens to be encoded using base64, but for e.g. the ProxyUserIdentityExtractor, the credential is simply what is passed through the Authorization header from the upstream proxy.